store/tikv/lock_test.go

```diff
@@ -126,6 +126,22 @@ func (s *testLockSuite) TestScanLockResolveWithSeek(c *C) {
 	}
 }
 
+func (s *testLockSuite) TestScanLockResolveWithSeekKeyOnly(c *C) {
+	s.putAlphabets(c)
+	s.prepareAlphabetLocks(c)
+
+	txn, err := s.store.Begin()
+	c.Assert(err, IsNil)
+	txn.SetOption(kv.KeyOnly, true)
+	iter, err := txn.Seek([]byte("a"))
+	c.Assert(err, IsNil)
+	for ch := byte('a'); ch <= byte('z'); ch++ {
+		c.Assert(iter.Valid(), IsTrue)
+		c.Assert([]byte(iter.Key()), BytesEquals, []byte{ch})
```

Comment:
Check value is empty?
And it should also turn `txn.SetOption(kv.KeyOnly, false)`  off to check that the `kv.KeyOnly` option take effect.

Reply:
This may need change mocktikv? @disksing

Reply:
> Check value is empty?

As @disksing said above, it can be 'undefined behavior' when setting the KeyOnly option

Reply:
The `txn.SetOption(kv.KeyOnly, false)` case is exactly what txn2 did, maybe it is not necessary to add another.

Reply:
I add another test case for KeyOnly as false